ext/serializer.c

```diff
@@ -821,13 +838,45 @@ static void _serialize_meta(zval *el, ddtrace_span_data *span) {
         ddtrace_exception_to_meta(Z_OBJ_P(exception_zv), meta, dd_add_meta_array, exception_type);
     }
 
-    zend_array *span_links_zv = ddtrace_spandata_property_links(span);
-    if (zend_hash_num_elements(span_links_zv) > 0) {
+    zend_array *span_links = ddtrace_spandata_property_links(span);
+    if (zend_hash_num_elements(span_links) > 0) {
         smart_str buf = {0};
-        _dd_serialize_json(span_links_zv, &buf, 0);
+        _dd_serialize_json(span_links, &buf, 0);
         add_assoc_str(meta, "_dd.span_links", buf.s);
     }
 
+    if (get_DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED()) { // opt-in
```

Comment:
shall this apply to the whole block or only for the automated `peer_service_sources` mapping? (i.e. line 858 currently)

Reply:
From my understanding of the RFC (Service Representation, part FAQ: Tag peer.service), I think this should apply to the whole block as I don't see anything mapping-related, as well as saying they "_would like to avoid releasing `peer.service` to the public and then have to think about a migration strategy_".

However, I would be more confident in this if @labbati also shares the same point of view 😃

Reply:
The whole block is fine.